src/datatable/datatable.jl

```diff
@@ -751,14 +751,9 @@ function hcat!(dt1::DataTable, dt2::AbstractDataTable)
     for i in 1:length(u)
         dt1[u[i]] = dt2[i]
     end
-
     return dt1
 end
-hcat!(dt::DataTable, x::CategoricalArray) = hcat!(dt, DataTable(Any[x]))
-hcat!(dt::DataTable, x::NullableCategoricalArray) = hcat!(dt, DataTable(Any[x]))
-hcat!(dt::DataTable, x::NullableVector) = hcat!(dt, DataTable(Any[x]))
-hcat!(dt::DataTable, x::Vector) = hcat!(dt, DataTable(Any[NullableArray(x)]))
```

Comment:
AFAICT this stops autopromotion to `NullableArray`? It's surprising that the tests didn't need adjustment. Please mention this in the commit message.